### PR TITLE
feat: feature gate c-kzg in reth-primitives

### PR DESCRIPTION
**Pull request by Rjected**
This feature gates `reth-primitives` types that require `c-kzg` behind a feature flag of the same name.

### CODE REVIEW

File: crates/primitives/Cargo.toml [hunk 1]
```diff
@@ -90,6 +90,7 @@ pprof = { version = "0.12", features = ["flamegraph", "frame-pointer", "criterio
 [features]
 default = []
```
Comment (mattsse) [inline crates/primitives/Cargo.toml:91]: should we make them default?

File: crates/primitives/src/constants/eip4844.rs [hunk 2]
```diff
@@ -34,45 +30,58 @@ pub const BLOB_TX_MIN_BLOB_GASPRICE: u128 = 1u128;
...
-        let _settings = Arc::clone(&MAINNET_KZG_TRUSTED_SETUP);
+        #[test]
+        fn ensure_load_kzg_settings() {
+            let _settings = Arc::clone(&MAINNET_KZG_TRUSTED_SETUP);
+        }
     }
 }
+
+#[cfg(feature = "c-kzg")]
+pub use trusted_setup::*;
```
Comment (mattsse) [inline crates/primitives/src/constants/eip4844.rs:87]: move above

File: crates/primitives/Cargo.toml (hunk 1 above)
Comment (Rjected) [reply crates/primitives/Cargo.toml:91]: sort of indifferent to what it should be tbh, but let's make it default for now

**Review by mattsse — APPROVED** [review]
lgtm

**Comment by mattsse** [comment]
good to merge @Rjected ?

**Pull request merged**